### PR TITLE
PERF: avoid double-verify of take indices

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing.html#code-standards) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/core/generic.py [hunk 1]
```diff
@@ -3655,6 +3655,7 @@ class  max_speed
 
         self._consolidate_inplace()
 
+        indices = extract_array(indices, extract_numpy=True)
```
Comment (jorisvandenbossche) [inline pandas/core/generic.py:3658]: Why is this needed here? The manager method has an `asarray(..)` call anyway (to convert other array-likes such as a list, or EAs, etc, so that will also convert any Index or Series passed through)

**Review by jorisvandenbossche — COMMENTED** [review]
Looks good! Small question on the added extract_array

File: pandas/core/generic.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/core/generic.py:3658]: leftover from an attempt at stronger annotating.  theres at least one test where an Index makes it through to the Block method.  will remove.

**Review by jorisvandenbossche — APPROVED** [review]

**Comment by jorisvandenbossche** [comment]
Thanks!
